### PR TITLE
More meaningful name for `@task.kubernetes` pods

### LINKED ISSUE #46464: More meaningful name for `@task.kubernetes` pods by default
### Description

Generate more meaningful default names for pods by `@task.kubernetes` with explicit randomness control

### Use case/motivation

Reference PR with tests showing the desired changes:
https://github.com/apache/airflow/pull/46462

Now pod created by decorator would have name `f"k8s_airflow_pod_{uuid.uuid4().hex}"` if name was not explicitly provided by the user.

I want more meaningful names by default (via python_callable name utilization) and to reuse already present randomness in name control option from `KubernetesPodOperator`: `random_name_suffix`.

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by insomnes**
Use name based on python callable in `@task.kubernetes` pod name generation 

- generate specific pod name base by provided python callable
- drop uuid usage with no name (random is controlled by `random_name_suffix` arg, there is no need for extra UUID usage)
- add specific pod naming tests for decorator flow because decorator distinguishes between `name=None` and no `name` argument provided

closes: #46464

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/decorators/__init__.pyi [hunk 1]
```diff
@@ -588,7 +588,7 @@ class TaskDecoratorCollection:
         :param name: Name of the pod to run. This will be used (plus a random
             suffix if *random_name_suffix* is *True*) to generate a pod ID
             (DNS-1123 subdomain, containing only ``[a-z0-9.-]``). Defaults to
```
Comment (RNHTTR) [inline airflow/decorators/__init__.pyi:590]: This docstring will need to be further updated, right?

**Review by RNHTTR — COMMENTED** [review]
What will happen if two Dag runs are triggered at the same time, and two `TaskInstance`s are created from the same task simultaneously?

**Comment by insomnes** [comment]
> What will happen if two Dag runs are triggered at the same time, and two `TaskInstance`s are created from the same task simultaneously?

If `random_name_suffix` arg is set (as default) they would have two different names by separate random suffixes the suffixes generated during `execute` calls in KPO by `build_pod_request_obj` method.

If user has disabled name randomization  by `random_name_suffix=False` -- that's more on user side to ensure name uniqueness.

File: airflow/decorators/__init__.pyi (hunk 1 above)
Comment (insomnes) [inline airflow/decorators/__init__.pyi:590]: I am sorry, but I don't get it.
What part of doc string should be updated or what should I add here? I believe current change is covered, what am I missing?

**Review by RNHTTR — APPROVED** [review]

**Comment by RNHTTR** [comment]
Looks good imo, but tests are failing now

**Comment by insomnes** [comment]
Oh, now there is conflict after the provider move to a new structure.  Will  check it a bit later today

**Comment by potiuk** [comment]
Yeah... Sorry :) . It was finaly green, so I merged it. But I think in this case git will properly fix stuff when you rebase -  automatically

**Comment by potiuk** [comment]
still conflicts to solve :(

**Comment by potiuk** [comment]
No worries :)
